src/aura/CRLP_FilterGroup/CRLP_FilterGroupHelper.js

```diff
@@ -69,24 +71,199 @@
+
+        if (filterRuleFieldType === "multipicklist" && constantName) {
+            updatedLabel = constantName.join(";\n");
+        } else if (filterRuleFieldType === "text" && constantName.indexOf(";") > 0 && constantName) {
+            var labelRe = /;[\n]+[ ]+|;[ ]+[\n]+|;/g;
+            updatedLabel = constantName.replace(labelRe, ";\n");
+
+            var nameRe = /\n| /g;
+            var newConstantName = constantName.replace(nameRe, "");
+            cmp.set("v.activeFilterRule.constantName", newConstantName);
+        }
+
+        return updatedLabel;
+    },
+
+    /**
+     * @description - changes picklist values or input type based on field type
+     * @param operator - selected filter rule operator API name
+     * @param type - DisplayType of the selected field transformed to lower case
+     */
+    rerenderValue: function(cmp, operator) {
+        var type = this.getFieldType(cmp, cmp.get("v.activeFilterRule.fieldName"));
+        console.log('type is ' + type);
+        console.log('operator is ' + operator);
+        if (type === 'boolean') {
+            //boolean fields don't have official translations (confirmed in process builder)
+            cmp.set("v.filterRuleFieldType", 'picklist');
+            var options = [{value: 'true', label: 'True'}, {value: 'false', label: 'False'}];
+            cmp.set("v.filterRuleConstantPicklist", options);
+        } else if (type === 'date') {
+            cmp.set("v.filterRuleFieldType", 'date');
+        } else if (type === 'datetime') {
+            cmp.set("v.filterRuleFieldType", 'datetime-local');
+        } else if (type === 'picklist' || type === 'multipicklist') {
+            if (operator === 'Equals' || operator === 'Not_Equals') {
+                cmp.set("v.filterRuleFieldType", 'picklist');
+            } else if (operator === 'Starts_With' || operator === 'Contains' ||  operator === 'Does_Not_Contain'){
+                cmp.set("v.filterRuleFieldType", 'text');
+            } else if (operator === 'In_List' || operator === 'Not_In_List'
+                || operator === 'Is_Included' || operator === 'Is_Not_Included'){
+                cmp.set("v.filterRuleFieldType", 'multipicklist');
+            }
+        } else if (type === 'reference') {
+            var field = cmp.get("v.activeFilterRule.fieldName");
+            //record type is a special case where we want to exactly match options and provide picklists
+            if (field !== 'RecordTypeId') {
+                cmp.set("v.filterRuleFieldType", 'text');
+            } else {
+                if (operator === 'Equals' || operator === 'Not_Equals') {
+                    cmp.set("v.filterRuleFieldType", 'picklist');
+                } else if (operator === 'In_List' || operator === 'Not_In_List') {
+                    cmp.set("v.filterRuleFieldType", 'multipicklist');
+                }
+            }
+        } else if (type === 'time') {
+            cmp.set("v.filterRuleFieldType", 'time');
```

Comment:
The filter rule logic will not support the new Time type. This code is probably ok, but we need to ensure that fields of type "Time" cannot be selected as a filter field.

Sorry, I completely forgot about this one.

Reply:
Will DateTime be accepted? Or just not Time?

Reply:
Yes. DateTime is technically OK (though I'm not sure I can find a use case), but Time is a new data type and I couldn't really come up with a use case that made it worth the time to build support for it.

Reply:
Time and Blob type (Large/Rich Text Area) are the only ones I can think of that would not be supported for filters. The reason I want to exclude Blob fields is that they have a negative impact on query performance, especially in a batch job. I can't really think of a good reason to support filtering on a 32K field anyway. Technically IsDeleted should be excluded as well because we never query deleted records, so including it could be confusing to users.

Reply:
Cool, I can include Blob no problem, and refactor a bit for IsDeleted. Thanks!